File: easy/sumtriangle/sumtriangle.c
/*
    started at: Saturday, 23. November 2013 05:33PM
*/

#include <stdio.h>


int max(int a,int b)
{
    return (a>b) ? a : b;
}

int main(int argc, char const *argv[])
{
    int triangle[100][100] = {{0}};
    int iterations = 0;
    int lines = 0;

    scanf("%d",&iterations);
    for (int it = 0; it < iterations; ++it)
    {
        scanf("%d",&lines);
        for (int i = 0; i < lines; ++i)
        {
            for (int j = 0; j <= i; ++j)
            {
                scanf("%d",&triangle[i][j]);
            }
        }

        //------------------------------

        int row = lines - 2;
        int col = 0;

        while(row>=0)
        {   col = 0;
            while(col<=row)
            {
                triangle[row][col] += max(triangle[row+1][col],triangle[row+1][col+1]);
                col++;
            }
            row--;
        }
        printf("%d\n",triangle[0][0]);
        
    }
    return 0;
}

File: easy/sumtriangle/sumtriangle.py
#!/usr/bin/env python
# -*- coding: utf-8 -*-

# started at: Saturday, 23. November 2013 05:33PM
import sys

def main():
    iterations = int(sys.stdin.readline())

    while iterations:
        lines = int(sys.stdin.readline())
        triangle = [[int(sys.stdin.readline())]]

        for i in xrange(1,lines):
            triangle.append(map(int,sys.stdin.readline().split()))


        row = len(triangle)-2 # second last

        while row>=0:
            col = 0
            while col<=row:  # length = row number
                triangle[row][col] += max(triangle[row+1][col],triangle[row+1][col+1])
                col +=1
            row -=1

        print triangle[0][0]
        iterations -= 1
    
if __name__ == '__main__':
    main()